cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/TemplateSetConfiguration.java

```diff
@@ -30,22 +42,111 @@ public class TemplateSetConfiguration {
   /** List of mavenCoordinates for the template sets that should be installed at cobigen startup */
   private List<MavenCoordinate> mavenCoordinates;
 
+  /** All available {@link Trigger}s */
+  private Map<String, Trigger> triggers;
+
+  /** All available {@Link Template}s */
+  private Map<String, Template> templates;
+
+  /** All available {@Link Increment} */
+  private Map<String, Increment> increments;
+
+  /** The templateSetConfiguration this decorator wraps */
+  private com.devonfw.cobigen.impl.config.entity.io.TemplateSetConfiguration templateSetConfiguration;
+
+  /** The reader to read the template-set.xml files */
+  public TemplateSetConfigurationReader templateSetConfigurationReader;
+
+  /** Paths of the template set configuration files */
+  public List<Path> templateSetFiles = new ArrayList<>();
```

Comment:
Is this needed in the class. The class is mainly used for reading and setting the properties file.

Reply:
Maybe we should split it into two classes. One just handles the properties and the other one is a data wraper for a TemplateSetConfiguration.

Reply:
✓